### DISCUSSION TITLE
Interpreting fiber's response during the ACTIVATION_THRESHOLD protocol

### DISCUSSION THREAD

**Discussion by PALMS-gui** [wmglab-duke/ascent#8] [Q&A] [answered]
I have done a simulation for ACTIVATION_THRESHOLD protocol and received _thresh_inner0_fiber0.dat_ file as well as _Vm_time_inner0_fiber0_amp0.dat_.
I assume that the second file represents the fiber response when given sample is stimulated by estimated threshold. 
Can you help me understand to which extent the protocol parameters may affect the response: 
```
 "threshold": {
      "value": Double,
      "n_min_aps": Integer,
      "ap_detect_location": Double
    }
```
Especially interested in the last one - at which location of the nerve to detect the AP? Is there a rule of thumb for this (e.g. the closest\furthest from the cuff electrodes?)

In my particular simulation I see the following Vm traces:
![afbeelding](https://user-images.githubusercontent.com/72300527/143905509-7b7281ea-6400-42d1-b5d6-6def2fed19e0.png)
that is 5 AP close by in the very beginning of the simulation and nothing afterwards, though stimulation is going on until 500 ms.

![afbeelding](https://user-images.githubusercontent.com/72300527/143907228-6eb0328e-aa35-4819-88f9-edef231aae1d.png)
How do I interpret this? 
Is the axons' refraction period so long (can't be, right?) or is it blocked? Trying to grasp the basics here, so thanks...

**Comment by ericmusselman** [comment, marked as answer]
Hi PALMS-gui

When running ACTIVATION_THRESHOLD, thresh_inner0_fiber0.dat contains the threshold for inner index #0 fiber index #0.

Vm_time_inner0_fiber0_amp0.dat contains Vm(t) at the locations specified in Sim->"saving"->"time"->"locs". Please see "JSON file parameter guide" in the ReadTheDocs documentation.

"ap_detect_location" needs to be far enough from the site of stimulation such that the change in Vm recorded is only from a passing action potential, rather than just an ohmic rise in transmembrane potential. The answer depends... generally in my independent work I use 0.1 or 0.9, but here are some important things to think about when determining this value: fiber model, fiber length, stimulation amplitudes, cuff position/length, ...

Since I do not know the details of your model, I cannot provide an interpretation of these plots. However, based on how you labeled the plot, it looks like you are plotting Vm(t) from each node, hence the slight delays between nodes for action potential conduction delay (across colors). To start, making a plot of Vm(x,t) for one fiber might be a good way to see what is going on.

Hope this helps!

**Reply by PALMS-gui** [reply]
Thank you for the reply!
in this simulation I am using a fiber diameter 10 stimulated with a 1000 Hz sinusoid.

here is how the raster plot (Vm(x,t)) with the threshold -40mV for one fiber looks like:
![afbeelding](https://user-images.githubusercontent.com/72300527/145273973-436cfc08-535f-4377-9802-a1ac9e0cc8e0.png)
and zoomed in 
![afbeelding](https://user-images.githubusercontent.com/72300527/145274268-796b57ee-6348-4b4c-9ff2-d240f008773e.png)

other fibers have similar pattern: one burst in the beginning and then nothing, except low-amplitude oscillations at stimulation frequency.
So there is really no other APs after that initial burst - the fact I can't yet explain. Among the four examples provided in the repo none is saving or analyzing Vm traces, so I can't run a validated experiment. I will try and change different model parameters one by one (cuff, stimulation type, AP record location, etc.) to see if this makes any difference.

Also, am I right that: the config "saving>>space>>vm" and "saving>>time>>vm" produce literally the same files, just transposed (in case all possible times and locs are selected?)

P.S.

Here is an example sim file I use:
`
{
  "n_dimensions": 2,
  "active_srcs": {
    "default": [[1, -1]]
  },
  "fibers": {
    "plot": false,
    "mode": "MRG_DISCRETE",
    "xy_trace_buffer": 50.0,
    "z_parameters": {
      "diameter": [1,2,10,14],
      "offset": 0,
      "seed": 123,
      "longitudinally_centered": true
    },
    "xy_parameters": {
      "mode": "CENTROID",
	  "count": 2,
      "top_down": true,
      "minimum_number": 1,
      "maximum_number": 20,
      "target_density": 0.0001,
      "target_number": 10,
      "seed": 123
    }
  },
  "waveform": {
    "plot": false,
    "global": {
      "unit": "ms",
      "dt": 0.05,
      "start": 0,
      "on": 1,
      "off": 500,
      "stop": 550
    },
    "SINUSOID": {
      "pulse_repetition_freq": [1000, 3000, 5000],
      "digits": 3
    }
  },
  "intracellular_stim": {
    "times": {
      "pw": 0,
      "IntraStim_PulseTrain_delay": 0,
      "IntraStim_PulseTrain_dur": 0
    },
    "pulse_repetition_freq": 0,
    "amp": 0,
    "ind": 2
  },
  "saving": {
    "space": {
      "vm": false,
      "gating": false,
      "times": [0]
    },
    "time": {
      "vm": true,
      "gating": false,
      "istim": false,
      "locs": "all"
    },
    "end_ap_times": {
      "loc_min": 0.1,
      "loc_max": 0.9,
      "threshold": -30
    },
    "runtimes": false
  },

    "protocol": {
    "mode": "ACTIVATION_THRESHOLD",
    "initSS": -200,
    "dt_initSS": 10,
    "threshold": {
      "value": -30,
      "ap_detect_location": 0.9,
      "n_min_aps": 1
    },
    "bounds_search": {
      "mode": "PERCENT_INCREMENT",
      "step": 5,
      "top": -1,
      "bottom": -0.1
    },
    "termination_criteria": {
      "mode": "PERCENT_DIFFERENCE",
      "percent": 1
    }
  },
  "supersampled_bases": {
    "generate": false,
    "use": false,
    "dz": 0.01,
    "source_sim": 1009
  }
}
`

**Reply by PALMS-gui** [reply]
When I use the same model\setup, but with FINITE_AMPLITUDES protocol and several amplitudes (close to and around) the threshold amplitude, recorded Vm files contain APs along the whole duration of the stimulation.

**Reply by ericmusselman** [reply]
I think what you are observing is the onset response of the fiber to 1 kHz stimulation. You might consider delivering a test pulse using intracellular stimulation to one end of the fiber (after the onset period is over) and check if the AP is blocked at the site of the extracellular stimulation cuff electrode. From personal experience, I suspect you'll get more reliable block at the 10-20 kHz range. You are correct about "saving>>space>>vm" and "saving>>time>>vm" saving the same data just saved in with different controls in a different format.

**Reply by PALMS-gui** [reply]
My simulations show that the fibers are not blocked, but simply the threshold found is just enough to stimulate one AP. I executed "amplitudes protocol" with a set of amplitudes slightly above previously derived thresholds and then starting with stimulation ~ threshold+10%, APs are generated uninterrupted along the whole timeline. 

I suspect if I used "n_min_aps" parameter higher than 1 (e.g. 4-5), that would help to have the found activation threshold a bit higher initially.  Though, n_min_aps=1 is the default in templates\manuals, which might be not the best choice (or at least a remark required).
